### PR TITLE
BLUEBUTTON-312: Create Platinum AMI Build Job

### PR DESCRIPTION
**Pull request by jzulim**
**1.** This PR supports the creation of a new pipeline job BUILD_PLATINUM_AMI which creates a 'Platinum' AMI based off a defined Gold AMI which follows the gold image and includes updates and BB app prereqs. This will shorten  

**2.** This PR also begins to create the new supporting BUILD_BB_AMI job which will be called downstream from the Platinum job to complete the agnostic AMI creation for our application instances. 

_Note: There is additional work to create the agnostic AMI but I wanted to include the basic components here so I can test within Jenkins in prep for a final push. These changes are separated from the current working build and deploy jobs. Those will not change and build / deploys will proceed as usual until Part 2 is completed, which is to modify the deploy jobs to support agnostic images. _

### CODE REVIEW

**Comment by johnfrenchxyz** [comment]
I don't know much about AMIs, so I'd love another dev to weigh in on this PR. Are there any particular factors or tests or things I should be looking at to test? I trust your dev work, but am just curious for my own edification.

**Comment by jzulim** [comment]
> I don't know much about AMIs, so I'd love another dev to weigh in on this PR. Are there any particular factors or tests or things I should be looking at to test? I trust your dev work, but am just curious for my own edification.

I appreciate your willingness to jump in here. I've fired off the build job using the new platinum config and am pasting a snippet of the logs here for show and tell. Also, you can use the AWS CLI (if configured for our environment, JF, you probably don't have this set because you don't do ops work, that's ok.) 

**The output from Jenkins:**
....
amazon-ebs: PLAY RECAP *********************************************************************
    amazon-ebs: default                    : ok=28   changed=24   unreachable=0    failed=0
    amazon-ebs:
==> amazon-ebs: Stopping the source instance...
    amazon-ebs: Stopping instance, attempt 1
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Creating the AMI: bb20-platinum-20190305040951
    amazon-ebs: AMI: ami-0fb97e53be29637ba
==> amazon-ebs: Waiting for AMI to become ready...
==> amazon-ebs: Adding tags to AMI (ami-0fb97e53be29637ba)...
==> amazon-ebs: Tagging snapshot: snap-059f1726e8d41a12b
==> amazon-ebs: Creating AMI tags
    amazon-ebs: Adding tag: "Name": "bb20-platinum-20190305040951"
==> amazon-ebs: Creating snapshot tags
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' finished.

==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs: AMIs were created:
us-east-1: ami-0fb97e53be29637ba

[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] }
[Pipeline] // withAwsCli
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Acquire Platinum AMI ID)
[Pipeline] script
[Pipeline] {
[Pipeline] dir
Running in /jenkins/blue-button-build-ami/code
[Pipeline] {
[Pipeline] sh
+ cut -d ' ' -f2
+ sed -e 's/ //g'
+ tr '
' ' '
+ grep 'us-east-1: ami-' platinum_packer_output.txt
[Pipeline] echo
BB20 PLATINUM AMI: ami-0fb97e53be29637ba

**AND THEN..
You can use AWS CLI to query our AMI's available:**

jzmbp:~ johnzulim$ aws --region us-east-1 ec2 describe-images --filters Name=name,Values='bb20-platinum-??????????????' --query 'sort_by(Images,&CreationDate)[-1]' --output json
{
    "Architecture": "x86_64",
    "CreationDate": "2019-03-03T20:22:29.000Z",
    "ImageId": "ami-06f615827c9522799",
    "ImageLocation": "501705132200/bb20-platinum-20190303081749",
    "ImageType": "machine",
    "Public": false,
    "OwnerId": "501705132200",
    "State": "available",
    "BlockDeviceMappings": [
        {
            "DeviceName": "/dev/sda1",
            "Ebs": {
                "DeleteOnTermination": true,
                "SnapshotId": "snap-057726c8653bc585c",
                "VolumeSize": 30,
                "VolumeType": "gp2",
                "Encrypted": false
            }
        }
    ],
    "EnaSupport": true,
    "Hypervisor": "xen",
    "Name": "bb20-platinum-20190303081749",
    "RootDeviceName": "/dev/sda1",
    "RootDeviceType": "ebs",
    "SriovNetSupport": "simple",
    "Tags": [
        {
            "Key": "Name",
            "Value": "bb20-platinum-20190303081749"
        }
    ],
    "VirtualizationType": "hvm"
}

File: Jenkinsfiles/Jenkinsfile.build_app_ami [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          fi
+        fi
+        """
+      }
+    }
+
+    stage('Set release_version var') {
+      steps {
+        script {
+          if (params.BB20_APP_VERSION == 'develop' || params.BB20_APP_VERSION == 'master') {
```
Comment (whytheplatypus) [inline Jenkinsfiles/Jenkinsfile.build_app_ami]: the reference to `develop` can be removed as it's not being used anymore.

File: Jenkinsfiles/Jenkinsfile.build_app_ami [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    stage('Checkout') {
+      steps {
+        checkout([
+          $class: 'GitSCM',
+          branches: [[
+            name: "${params.BB20_DEPLOY_BRANCH}"
+          ]],
+          doGenerateSubmoduleConfigurations: false,
```
Comment (whytheplatypus) [inline Jenkinsfiles/Jenkinsfile.build_app_ami]: possibly need this to be `true` now that the openapi definition is pulled in through a submodule

**Review by whytheplatypus — CHANGES_REQUESTED** [review]

File: Jenkinsfiles/Jenkinsfile.build_app_ami (hunk 2 above)
Comment (jzulim) [reply Jenkinsfiles/Jenkinsfile.build_app_ami]: Nice catch. Thank you.

**Review by whytheplatypus — APPROVED** [review]

**Pull request merged**